### PR TITLE
[SPARK-30812][SQL][CORE] Revise boolean config name to comply with new config naming policy

### PR DESCRIPTION
**Pull request by Ngone51**
### What changes were proposed in this pull request?

Revise below config names to comply with [new config naming policy](http://apache-spark-developers-list.1001551.n3.nabble.com/DISCUSS-naming-policy-of-Spark-configs-td28875.html):

SQL:
* spark.sql.execution.subquery.reuse.enabled / [SPARK-27083](https://issues.apache.org/jira/browse/SPARK-27083)
* spark.sql.legacy.allowNegativeScaleOfDecimal.enabled / [SPARK-30252](https://issues.apache.org/jira/browse/SPARK-30252)
* spark.sql.adaptive.optimizeSkewedJoin.enabled / [SPARK-29544](https://issues.apache.org/jira/browse/SPARK-29544)
* spark.sql.legacy.property.nonReserved / [SPARK-30183](https://issues.apache.org/jira/browse/SPARK-30183)
* spark.sql.streaming.forceDeleteTempCheckpointLocation.enabled / [SPARK-26389](https://issues.apache.org/jira/browse/SPARK-26389)
* spark.sql.analyzer.failAmbiguousSelfJoin.enabled / [SPARK-28344](https://issues.apache.org/jira/browse/SPARK-28344)
* spark.sql.adaptive.shuffle.reducePostShufflePartitions.enabled / [SPARK-30074](https://issues.apache.org/jira/browse/SPARK-30074)
* spark.sql.execution.pandas.arrowSafeTypeConversion / [SPARK-25811](https://issues.apache.org/jira/browse/SPARK-25811)
* spark.sql.legacy.looseUpcast / [SPARK-24586](https://issues.apache.org/jira/browse/SPARK-24586)
* spark.sql.legacy.arrayExistsFollowsThreeValuedLogic / [SPARK-28052](https://issues.apache.org/jira/browse/SPARK-28052)
* spark.sql.sources.ignoreDataLocality.enabled / [SPARK-29189](https://issues.apache.org/jira/browse/SPARK-29189)
* spark.sql.adaptive.shuffle.fetchShuffleBlocksInBatch.enabled / [SPARK-9853](https://issues.apache.org/jira/browse/SPARK-9853)

CORE:
* spark.eventLog.erasureCoding.enabled / [SPARK-25855](https://issues.apache.org/jira/browse/SPARK-25855)
* spark.shuffle.readHostLocalDisk.enabled / [SPARK-30235](https://issues.apache.org/jira/browse/SPARK-30235)
* spark.scheduler.listenerbus.logSlowEvent.enabled / [SPARK-29001](https://issues.apache.org/jira/browse/SPARK-29001)
* spark.resources.coordinate.enable / [SPARK-27371](https://issues.apache.org/jira/browse/SPARK-27371)
* spark.eventLog.logStageExecutorMetrics.enabled / [SPARK-23429](https://issues.apache.org/jira/browse/SPARK-23429)

### Why are the changes needed?

To comply with the config naming policy.

### Does this PR introduce any user-facing change?

No. Configurations listed above are all newly added in Spark 3.0.

### How was this patch tested?

Pass Jenkins.

### CODE REVIEW

**Comment by Ngone51** [comment]
Note that I tried to list all the configs which break the policy from year 2019 rather than from Spark 2.0.

**Review by dongjoon-hyun — COMMENTED** [review]
Oh, is this backward compatible, @Ngone51 ? I'm wondering all of these are new configurations added at 3.0.0.

**Comment by jiangxb1987** [comment]
retest this please

**Comment by Ngone51** [comment]
> Oh, is this backward compatible, @Ngone51 ? I'm wondering all of these are new configurations added at 3.0.0.

@dongjoon-hyun We're targeting at configs added in 3.0 only without any backward compatible maintain in this PR.

I'll check them again for sure.

**Comment by Ngone51** [comment]
Hi @dongjoon-hyun , I've checked again and attached all the related JIRAs.

**Comment by Ngone51** [comment]
also cc @cloud-fan

File: core/src/main/scala/org/apache/spark/internal/config/package.scala [hunk 1]
```diff
@@ -143,7 +143,7 @@ package object config {
       .createWithDefault(false)
 
   private[spark] val EVENT_LOG_ALLOW_EC =
-    ConfigBuilder("spark.eventLog.erasureCoding.enabled")
+    ConfigBuilder("spark.eventLog.allowErasureCoding")
```
Comment (cloud-fan) [inline core/src/main/scala/org/apache/spark/internal/config/package.scala]: `allow` looks weird, I'd keep this unchanged or `enableErasureCoding`. Do we expect to see more configs starts with `spark.eventLog.erasureCoding`? It seems possible if we want some fine-grained controls over how to do erasure coding.

File: core/src/main/scala/org/apache/spark/internal/config/package.scala [hunk 2]
```diff
@@ -1114,16 +1114,6 @@ package object config {
       .booleanConf
       .createWithDefault(false)
 
-  private[spark] val STORAGE_LOCAL_DISK_BY_EXECUTORS_CACHE_SIZE =
-    ConfigBuilder("spark.storage.localDiskByExecutors.cacheSize")
```
Comment (cloud-fan) [inline core/src/main/scala/org/apache/spark/internal/config/package.scala]: good catch! It's weird to see a storage config in the middle of a bunch shuffle configs.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 3]
```diff
@@ -2142,7 +2142,7 @@ object SQLConf {
       .createWithDefault(false)
 
   val LEGACY_PROPERTY_NON_RESERVED =
-    buildConf("spark.sql.legacy.property.nonReserved")
+    buildConf("spark.sql.legacy.property.nonReserved.enabled")
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: nit: `spark.sql.legacy.allowReservedProperties` ?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 4]
```diff
@@ -2100,7 +2100,7 @@ object SQLConf {
       .stringConf
       .createOptional
 
-  val LEGACY_LOOSE_UPCAST = buildConf("spark.sql.legacy.looseUpcast")
+  val LEGACY_LOOSE_UPCAST = buildConf("spark.sql.legacy.looseUpcast.enabled")
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: `spark.sql.legacy.doLooseUpcast`? Then we can avoid creating the `looseUpcast` namespace.

File: core/src/main/scala/org/apache/spark/internal/config/package.scala (hunk 1 above)
Comment (Ngone51) [reply core/src/main/scala/org/apache/spark/internal/config/package.scala]: Ok. I'll keep it unchanged.

**Comment by Ngone51** [comment]
Jenkins, retest this please.

File: docs/configuration.md [hunk 5]
```diff
@@ -1641,7 +1641,7 @@ Apart from these, the following properties are also available, and may be useful
 <table class="table">
 <tr><th>Property Name</th><th>Default</th><th>Meaning</th></tr>
 <tr>
-  <td><code>spark.eventLog.logStageExecutorMetrics.enabled</code></td>
+  <td><code>spark.eventLog.logStageExecutorMetrics</code></td>
```
Comment (cloud-fan) [inline docs/configuration.md]: @HyukjinKwon is it possible to generate core config document automatically as well?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 6]
```diff
@@ -425,7 +425,7 @@ object SQLConf {
     .createWithDefault(true)
 
   val ADAPTIVE_EXECUTION_SKEWED_JOIN_ENABLED =
-    buildConf("spark.sql.adaptive.optimizeSkewedJoin.enabled")
+    buildConf("spark.sql.adaptive.optimizeSkewedJoin")
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: `optimizeSkewedJoin` is used as a namespace and has sub-configs. how about `spark.sql.adaptive.skewedJoinOptimization.enabled`?

cc @JkSelf @maryannxue

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 7]
```diff
@@ -1629,7 +1629,7 @@ object SQLConf {
       .createWithDefault(true)
 
   val PANDAS_ARROW_SAFE_TYPE_CONVERSION =
-    buildConf("spark.sql.execution.pandas.arrowSafeTypeConversion")
+    buildConf("spark.sql.execution.pandas.arrowSafeTypeConversion.enabled")
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: `arrowSafeTypeConversion` does not have sub-configs, is it possible to make it a verb?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 8]
```diff
@@ -2115,7 +2115,7 @@ object SQLConf {
     .createOptional
 
   val LEGACY_ARRAY_EXISTS_FOLLOWS_THREE_VALUED_LOGIC =
-    buildConf("spark.sql.legacy.arrayExistsFollowsThreeValuedLogic")
+    buildConf("spark.sql.legacy.arrayExistsFollowsThreeValuedLogic.enabled")
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: ditto, would be better to make it a verb. e.g. `followThreeValuedLogicInArrayExists`

**Comment by cloud-fan** [comment]
FYI I've open the PR to document naming guidelines: https://github.com/apache/spark/pull/27577

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala (hunk 6 above)
Comment (JkSelf) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: +1 for `spark.sql.adaptive.skewedJoinOptimization.enabled` to apply the naming guidelines.

**Comment by maropu** [comment]
> Hi @dongjoon-hyun , I've checked again and attached all the related JIRAs.

I've double-checked that; all the boolean options below are generated by [running code](https://gist.github.com/maropu/3054150bf16aa10795d52b48c3eaa921) in v2.4.5/master. I've checked that all the boolean configs in this PR are included in the list below;
```
// All the boolean SQL options newly added in v3.0.0
+--------------------------------------------------------------+                
|_c0                                                           |
+--------------------------------------------------------------+
|spark.sql.adaptive.forceApply                                 |
|spark.sql.adaptive.optimizeSkewedJoin.enabled                 |
|spark.sql.adaptive.shuffle.fetchShuffleBlocksInBatch.enabled  |
|spark.sql.adaptive.shuffle.localShuffleReader.enabled         |
|spark.sql.adaptive.shuffle.reducePostShufflePartitions.enabled|
|spark.sql.analyzer.failAmbiguousSelfJoin.enabled              |
|spark.sql.ansi.enabled                                        |
|spark.sql.cbo.planStats.enabled                               |
|spark.sql.codegen.aggregate.map.vectorized.enable             |
|spark.sql.codegen.aggregate.splitAggregateFunc.enabled        |
|spark.sql.csv.filterPushdown.enabled                          |
|spark.sql.datetime.java8API.enabled                           |
|spark.sql.defaultUrlStreamHandlerFactory.enabled              |
|spark.sql.execution.arrow.sparkr.enabled                      |
|spark.sql.execution.pandas.arrowSafeTypeConversion            |
|spark.sql.execution.subquery.reuse.enabled                    |
|spark.sql.inMemoryTableScanStatistics.enable                  |
|spark.sql.jsonGenerator.ignoreNullFields                      |
|spark.sql.legacy.addDirectory.recursive.enabled               |
|spark.sql.legacy.allowNegativeScaleOfDecimal.enabled          |
|spark.sql.legacy.arrayExistsFollowsThreeValuedLogic           |
|spark.sql.legacy.bucketedTableScan.outputOrdering             |
|spark.sql.legacy.createHiveTableByDefault.enabled             |
|spark.sql.legacy.dataset.nameNonStructGroupingKeyAsValue      |
|spark.sql.legacy.exponentLiteralAsDecimal.enabled             |
|spark.sql.legacy.fromDayTimeString.enabled                    |
|spark.sql.legacy.integralDivide.returnBigint                  |
|spark.sql.legacy.json.allowEmptyString.enabled                |
|spark.sql.legacy.looseUpcast                                  |
|spark.sql.legacy.property.nonReserved                         |
|spark.sql.legacy.sessionInitWithConfigDefaults                |
|spark.sql.legacy.setCommandRejectsSparkCoreConfs              |
|spark.sql.legacy.timeParser.enabled                           |
|spark.sql.legacy.typeCoercion.datetimeToString.enabled        |
|spark.sql.optimizer.dynamicPartitionPruning.enabled           |
|spark.sql.optimizer.dynamicPartitionPruning.reuseBroadcast    |
|spark.sql.optimizer.dynamicPartitionPruning.useStats          |
|spark.sql.optimizer.expression.nestedPruning.enabled          |
|spark.sql.optimizer.serializer.nestedSchemaPruning.enabled    |
|spark.sql.orc.mergeSchema                                     |
|spark.sql.sources.ignoreDataLocality.enabled                  |
|spark.sql.sources.validatePartitionColumns                    |
|spark.sql.streaming.checkpoint.escapedPathCheck.enabled       |
|spark.sql.streaming.fileSource.schema.forceNullable           |
|spark.sql.streaming.forceDeleteTempCheckpointLocation.enabled |
|spark.sql.streaming.stopActiveRunOnRestart                    |
|spark.sql.streaming.ui.enabled                                |
+--------------------------------------------------------------+
```

**Comment by Ngone51** [comment]
@maropu Thanks for the double check.  It's awesome! Probably, we could add some missing configs from the list into this PR.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala (hunk 7 above)
Comment (Ngone51) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: updated with `convertToArrowArraySafely`.

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
retest this please

**Comment by cloud-fan** [comment]
retest this please

**Comment by cloud-fan** [comment]
thanks, merging to master/3.0!

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 9]
```diff
@@ -425,7 +425,7 @@ object SQLConf {
     .createWithDefault(true)
 
   val ADAPTIVE_EXECUTION_SKEWED_JOIN_ENABLED =
-    buildConf("spark.sql.adaptive.optimizeSkewedJoin.enabled")
+    buildConf("spark.sql.adaptive.skewedJoinOptimization.enabled")
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: ah I missed this one. We should update other skew join related configs to use the `skewedJoinOptimization` prefix.

@JkSelf can you help to fix it?

**Comment by Ngone51** [comment]
thanks all!!

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala (hunk 9 above)
Comment (JkSelf) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: fix it in [PR#27630](https://github.com/apache/spark/pull/27630).

**Pull request closed without merging**